core/execution_state.cpp

```diff
@@ -130,13 +130,12 @@ void ExecutionState::addReachabilityMapping(const IR::Node *node, const IR::Expr
         return;
     }
 
-    if (!reachabilityMap.initializeReachabilityMapping(
+    if (!reachabilityMap.initializeOrAddReachabilityMapping(
```

Comment:
Things are a little tricky here but I would simply make the reachability map a vector of reachabilityExpressions and then append if a node already exists. Does this cause problems somewhere? I really want to make sure the ReachabilityExpression stays at is, because they are already tricky to use.

Reply:
The problem I have is not sure where to store the computed reachability in `computeNodeReachability`. So I wrapped the `vector<ReachabilityExpressions>` with `OredReachabilityExpression`.

Reply:
You can still compute the reachability individually for each expression. The only time we need to OR it is when we access in the dead code eliminator, no?

Reply:
Ah, you are right. Will fix it soon.